.github/workflows/integration_tests.yaml

```diff
@@ -2,24 +2,31 @@
 name: Integration tests
 on:
   pull_request_review:
-    types: [approved]
+    types: [submitted]
   workflow_dispatch:  # <-- can be triggered manually!
 
 jobs:
   integration-tests:
+    if: github.event.review.state == 'approved'
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macos-13]
         python-version: [3.10, 3.11, 3.12]
```

Comment:
It should be sufficient if we just grab the minimum version and max similar to [python-RAT](https://github.com/RascalSoftware/python-RAT/blob/main/.github/workflows/run_tests.yml) and change the values to strings i.e. "3.10" instead of 3.10 because GitHub is printing 3.1 in the output